<?xml version="1.0" encoding="UTF-8" standalone="no" ?>
<launch>
	<arg name="model_file" 	default="/srv/data/gait1992/residual_reduction_algorithm/model_adjusted.osim" />
	<arg name="get_second_label" default="true"/>
	<arg name="n_proc" default="8"/>
	<arg name="moment_arm_library_path"	default="/catkin_opensim/devel/lib/Gait1992MomentArm" />
	<!--arg name="moment_arm_library_path"	value="/catkin_ws/devel/lib/Gait19xxMomentArm" / -->

        <node pkg="osrt_ros" type="so_rr" 	name="so_rr_node" output="screen">
                <param name="model_file" 		value="$(arg model_file)" />
                <param name="get_second_label" 		value="$(arg get_second_label)" />
                <param name="n_proc" 			value="$(arg n_proc)" />

		<param name="convergence_tolerance" 	value="1.5e-0" />
		<param name="memory_history" 		value="10" />
		<param name="maximum_iterations" 	value="50" />
		<param name="objective_exponent" 	value="2" />
		<!-- there were some nice env variables defined to locate this, but now we have to specify explicitly -->
		<param name="moment_arm_library_path"	value="$(arg moment_arm_library_path)" />
		
		<remap from="/so_rr_node/sync_input" to="/id_node/output_multi"/>
		<remap from="/in_labels" to="/ik/outlabels"/>
		<!--remap from="/so_rr_node/input2" 	to="/inverse_kinematics_from_file/r_data2"/>
		<remap from="/in_labels2" 	to="/inverse_kinematics_from_file/out_labels"/ -->
		<!-- remap from="/so_rr_node/input2" 	to="/id_node/output"/ -->
		<remap from="/in_labels2" 	to="/id_node/out_labels"/>
		
        </node>

</launch>
